drop-in/src/main/java/com/adyen/checkout/dropin/internal/ui/DropInViewModel.kt

```diff
@@ -63,7 +63,7 @@ internal class DropInViewModel(
     val serviceComponentName: ComponentName = requireNotNull(bundleHandler.serviceComponentName)
 
     var amount: Amount
-        get() = bundleHandler.amount ?: getInitialAmount()
+        get() = requireNotNull(bundleHandler.amount)
```

Comment:
It shouldn't, but I have no way to verify it 100% (i.e. tests 😭) except by manually testing all scenarios, which I did.

To explain the change, previously we were only storing the amount in the bundle when we modify it on purpose like in the partial payment flow. But in the normal flow we would read `bundleHandler.amount` which would return null and then we'll call `getInitialAmount()` which returns `sessionDetails?.amount ?: dropInConfiguration.amount`. That created a problem for me because I didn't have access to the correct amount in the `DropInViewModelFactory` which I needed for the analytics.

So what I did instead in [this commit](https://github.com/Adyen/adyen-android/pull/1282/commits/3287e24253c366b2c398229c3c7ee20e94bf5b6d), is to set the amount in the bundle already before launching drop-in, the value would be `dropInConfiguration.amount` in a non sessions flow and `checkoutSession.sessionSetupResponse.amount` in a sessions flow. And then `DropInViewModel` and `DropInViewModelFactory` would only read that value from the bundle.

WDYT?

Reply:
Ohh makes sense now 👍